File: map/Interpreter2.0/DataStructure/IntFileTable.cs
using System;
using System.IO;
namespace DataStructure
{
    public interface IntFileTable
    {
        bool file_is_open(string file_name);
        int add(Tuple<String, StreamReader> val);
        void remove(int key);
        Tuple<String, StreamReader> get(int key);
        void update(int key, Tuple <String, StreamReader> val);
        bool exists(int key);
    }
}

File: map/InterpreterFX/src/DataStructure/IntFileTable.java
package DataStructure;

import ExceptionHandling.MyException;


import java.io.BufferedReader;
import java.util.Map;

public interface IntFileTable{
    boolean file_is_open(String file_name);
    int add(Pair<String, BufferedReader> val);
    void remove(int key);
    Pair<String, BufferedReader> get(int key) throws MyException;
    void update(int key, Pair <String, BufferedReader> val) throws MyException;
    boolean exists(int key);
    Map<Integer, Pair<String, BufferedReader>> getContent();
}
